### DISCUSSION TITLE
I'm very confused about AppConfig

### DISCUSSION THREAD

**Discussion by isaackearl** [nuxt/nuxt#19278] [Questions]
So I started a new small project, and when I run npm run generate it does an amazing thing, and it creates all the output and one of the output files in the _nuxt direction is an app.config file, where it has all my app config values.  This is really convenient for being able to change configurations for a staticly deployed app.

I have a larger app where this file has inexplicably stopped generating and I have no idea why.  If I make a new project it does it properly, but on my larger project none of the chunks are the app.config file... The app still works, but the problem is that now I can't modify the values in that config after generating the build.  I'm at a complete loss as to what the difference is between my small project and my larger one.

any help would be appreciated.

Also as a note: I cannot use node for deployment so using runtime configurations is not an option.

**Comment by isaackearl** [comment]
Follow question: Is there anyway to use app.baseUrl as an appConfig variable instead of runtimeConfig?  Very limiting being runtime only since runtime configurations don't seem to work with npm run generate (non-node hosting).

**Reply by TheAlexLichter** [reply]
Runtime config should work fine for any kind of hosting. You have to ensure to make the environment variables available though "on your own". Nuxt only reads them from the actual present env variables in production, **not** from a `.env` file (as in dev)

**Reply by isaackearl** [reply]
> Runtime config should work fine for any kind of hosting. You have to ensure to make the environment variables available though "on your own". Nuxt only reads them from the actual present env variables in production, **not** from a `.env` file (as in dev)

It doesn't work as far as I can tell when staticly generating the code (npm run generate).  I've tried on mac and windows machines.  I've set environment variables on the machine I'm on via export NUXT_ etc etc... I've tried all manner of things for 2 days straight at this point.  

Last night I made a new small repository and I tested a bunch of things.  The only configuration I could get environment variables working for is when using npm run build (building it for a node deployment).  I tried ssr: false, I tried ssr:true etc.  Anytime I use npm run generate, I can no longer get the project to respond to env variables.  I kill the process npm run preview, and I run it again and changes to environment variables don't do anything.

I would be willing to hire somebody to consult me on this if it is true that it works.  I'm pretty desperate at this point.

**Reply by TheAlexLichter** [reply]
Well, after static site generation, the files are static and you cannot provide a dynamic configuration anymore.

**Reply by isaackearl** [reply]
> Well, after static site generation, the files are static and you cannot provide a dynamic configuration anymore.

Yes that's the part that is confusing.  It doesn't really say that in the docs clearly.  There are a lot of other tools out there where you can do runtime configurations with the artifact that you generate for the site.  But I mostly come from a .Net framework world... At my current job, every tool that we use has the ability to do run time configurations after static generation... And our deployment options are very limited because of the circumstances of our job and where we deploy.  

Typically, there are ways to modify underlying values such as web.config options etc.

That being said, when I statically generate a site and I use an appConfig for my variables, It actually outputs the files in a way where I can modify the values in the static content... And therefore I can change between each time that I host it...  In other words, I'm finding that using appConfig is very convenient for statically generated sites and being able to modify runtime variables.

My problem is now that baseUrl is a runtime configuration and I can't modify that... And my second problem is that the outputted files from the generate function. I have to dig through them in order to find the appConfig.  It's not a huge problem, but it would be nice if there was a way to force the app config chunk to be on its own rather than inside of an entry file.  

When I make a new project with a very simple page it does just that by default... It's really nice.  But in my larger project that I've already spent several weeks on, it's buried in an entry.hash.js file...

I would love to be able to control this so it makes a chunk just for the app config file.  

Any help or guidance or tips or anything like that would be really appreciated.  When I get to my hotel tonight I'll probably try and provide some screenshots of what I'm talking about.

Thanks

**Reply by isaackearl** [reply]
> Runtime config should work fine for any kind of hosting. You have to ensure to make the environment variables available though "on your own". Nuxt only reads them from the actual present env variables in production, **not** from a `.env` file (as in dev)

To clarify, it is IIS hosting without a node executable (no iisnode or anything).  That means only using npm run generate works for our deployment.  

Would it be possible as a "feature request" to change how the bundle works so it puts the app.config files in it's own chunk?  Also it would be nice if baseUrl could be configured to be an appConfig variable as well.
